### ISSUE TITLE
Escorts moved via mini-map missing from shipyard

### ISSUE THREAD

**Issue by MikeDP** [endless-sky/endless-sky#9510]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Move selected escorts via a galaxy map/mini-map right-click, to system A (with shipyard).  Main fleet then lands on B.  Using the player/ship info panel, **_park_** the remote escorts on A.  Fly main fleet from B to A and land. 

Now, when you visit the shipyard/outfitters, the initially moved escorts do not appear in the ships list.  If you **_unpark_** them, then **depart and immediately land**, they are now visible in the shipyard/outfitters.

### Steps to Reproduce

1. Go to '_Beta Lupi'_ and capture pirates (which become 'Escorts X').
2. Use galaxy/mini-map and right-click to send escorts X to '_Seginus/Clark'_
3. Jump fleet to '_Men'_.  Wait for escorts X to reach '_Seginus'_.  Land on '_Thule'_. 
4. On Player/Ship Info screens, park escorts X at '_Seginus'_.
5. Depart 'Thule', jump to '_Seginus'_ and land on '_Clark'_.
6. In Shipyard/Outfitter screens escorts X are **not shown**.
7. In Player/Ship Info, unpark escorts X.
8. In Shipyard/Outfitter screens escorts X are **still not shown**.
9. Depart and immediately land again.
10. Escorts X are now visible in Shipyard and outfitters

### Expected Behavior

Ships remotely sent to other systems and parked should be immediately visible in Shipyard/Outfitters when main fleet arrives.

### Operating System

Linux Mint 20.3 Una

### Game Source

GitHub Releases

### Game Version

v 0.10.4 appimage downloaded 23/10/23

### Additional Information

Issue seen on laptop and desktop machines.  Same OS/game version on both.

**Comment by TomGoodIdea** [comment]
The fleet list doesn't say on which planet, only in which system your ships are located. That means there's no visual difference between ships parked _on a planet_ and those parked _in a system, but not on any planet_. That, combined with the fact that your escorts in other systems don't actually land when you do, causes the problem you've described.

**Comment by MikeDP** [comment]
I picked Seginus/Clark specifically as it has only one planet on which to land, so there is no ambiguity over where they could have landed.

OK. I take your point.  I had not considered _parking in a system_ as opposed to _on a planet_ as an option.

However, now the problem becomes that they _**behave** as if they have landed on Clark!_  

When I move my fleet to Clark, land and _unpark_ and then depart, the ships in question do appear (visually) to depart the planet (and not streak in from somewhere deep-system), **and** they have been refuelled, hulls and shields replenished (100%)

It's perhaps not an issue, merely an inconsistency. :wink:

**Comment by MikeDP** [comment]
OK.  I've had a look in the save file and I can see you are correct about "parking".  No planet is assigned to the "remotely moved and parked" ships e.g.
```
     ...
     system Seginus
     parked
```
whereas ships I've explicitly parked look like:
```
     ...
     system Sol
     planet Luna
     parked
```
Thus, the odd behaviour is the ships actually looking and behaving like they had actually _landed on a planet._ 

So, should we close this?

**Comment by TomGoodIdea** [comment]
> Thus, the odd behaviour is the ships actually looking and behaving like they had actually _landed on a planet_.
>
> So, should we close this?

No, because this behavior is confusing and clearly it's a bug (idk what exactly causes it, yet).

**Comment by MikeDP** [comment]
OK.  I'll spend a little more time looking into specific cases including "corner cases" - what happens if I remote to a pirate/restricted/hostile/uninhabited  etc.

**Comment by oldpenguin5** [comment]
the above mentioned fix to #9547 **does not solve the problem**. If  a ship in another system is ordered to park via 'View Player Info (I)', and there are planets it can actually land on, it should actually land with the priority : with shipyard/with outfitter/with spaceport/other 

Otherwise you can now easily get stuck:

1. in system A (with a planet with an outfitter), disable an enemy ship
2. board it, take its cooling equipment (so that it would overheat if departing from a planet - which you do not realize at the moment)
3. then try to capture it - and succeed in capturing it (it now belongs to your fleet)
4. tell the captured ship to hold position and jump to system B (or with your jump drive jump to system B which the ship cannot reach)
5. land on system B
6. depart - you get the following message:
![image](https://github.com/user-attachments/assets/adb4a885-eeff-4e11-8cc7-7f8f20ba5482)
7. if you press cancel, your departure is canceled; if you depart again, your'e back to step 6.
8. if you press Ok, the ship is parked in A (like if you parked it via 'I') and you can leave.
9. if you now go to system A and land there (because you want to reinstall the cooling equipment in the captured ship), your'e stuck because:
- the captured ship does not show up in the Outfitter (because it is 'parked in space'), so you cannot install the cooling outfit again
- when you try the workaround mentioned above in https://github.com/endless-sky/endless-sky/issues/9510#issuecomment-1781589147 and depart, you get the message 
![image](https://github.com/user-attachments/assets/e667ac44-38b6-4102-9929-d3693f3240f3) 
and are blocked from departing (even though the overheating ship is  not departing but already 'in space')
- the captured ship is practically lost - you can only park it again 'in space'  or disown it  

An alternative solution would be that if a ship is 'parked in space' in a system, and you land  on one of the system's planets, you unpark the ship and then park it again,  it will be parked on the planet you're on (and not 'in space' again); this could also apply to ships parked on another panet in the system: unparking them and parking them again would move them to the planet you're on.

**Comment by Mernom** [comment]
If you enable escort overlays, you can see the overlay ghosts when you enter a system where escorts were parked while not landed.
